### PR TITLE
added authentication for login/signup

### PR DESCRIPTION
**Pull request by malabudi**
Used next auth for login/signup, nextauth will also be used to log out

### CODE REVIEW

File: src/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Without a defined matcher, this line applies next-auth to the entire project
+export { default } from 'next-auth/middleware';
+
+// Pages that the user cannot visit unless they login
+export const config = {
+  matcher: ['/', '/capture', '/maps', '/settings'],
```
Comment (ms-977) [inline src/middleware.ts:21]: add  /editaccount

**Review by ms-977 — CHANGES_REQUESTED** [review]
suggested one small change

File: src/middleware.ts (hunk 1 above)
Comment (malabudi) [reply src/middleware.ts:21]: those are sub pages of settings, so the URL would look like these for the edit account pages

`settings/edit-email`
`settings/change-password`

Comment (malabudi) [reply src/middleware.ts:21]: with the current code, everything under `settings/` will naturally be a protected page

**Review by ms-977 — APPROVED** [review]

**Review by ms-977 — APPROVED** [review]

**Pull request merged**